src/php/Formatter/FormatterFacadeInterface.php

```diff
@@ -0,0 +1,10 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Phel\Formatter;
+
+interface FormatterFacadeInterface
```

Comment:
Do we really need interfaces for each Facade? Could it, in theory, be possible to have multiple implementations of this interface?

Reply:
The goal of the interface for the Facade is not to have multiple implementations for it but to **invert the dependencies** between the modules, so one module can communicate with another module via their Facade signatures/abstraction (interface). 
In short: the interfaces for the Facades are inverting the dependencies between the ones that used them.

Reply:
Ok. However, the PrinterFacade is not using an Interface. Did you missed it or is there another reason?

Reply:
Ahh, yes, that's right. The thing is that currently the `PrinterFacade` is not being used at all.
We are currently still using the old concept behavior of the `Printer`, using the static `readable()` and `nonReadable()`. 

So, this is a really good point/catch. My first idea was using the `PrinterFacade` as well (similar philosophy as every other module) but, the `Printer` is "coupled" somehow to the "Lang module" in a way which I preferred to work on in a separate PR (in order to simplify this PR a bit).
This is the reason why there is no PrinterFacadeInterface, basically, because the PrinterFacade is not in used (at least currently). I plan to work on it in the next iteration. So we can do two things:
1) I remove the unused code (PrinterFacade and PrinterFactory), and I create them in the future PR
2) I leave these two classes and I will continue with them in the next PR

I would prefer to do the 1) option. Is this ok for you, @jenshaase?

Reply:
I would suggest 2. and also add the interface. Then we have a complete Printer Module and can refactor the usage of the Printer in another PR.

Reply:
Ok, then we leave it the second option. Totally ok for me as well. Then there is nothing else to add here :)